### PR TITLE
Remove padding in batched generation.

### PR DESCRIPTION
**Pull request by lvwerra**
Remove padding in batched generation as a post-processing step.

### CODE REVIEW

File: examples/sentiment/scripts/gpt2-sentiment.py [hunk 1]
```diff
@@ -60,13 +60,14 @@ class ScriptArguments:
     model_name: Optional[str] = field(default="lvwerra/gpt2-imdb", metadata={"help": "the model name"})
     log_with: Optional[str] = field(default=None, metadata={"help": "use 'wandb' to log with wandb"})
     learning_rate: Optional[float] = field(default=1.41e-5, metadata={"help": "the learning rate"})
-    mini_batch_size: Optional[int] = field(default=128, metadata={"help": "the PPO minibatch size"})
-    batch_size: Optional[int] = field(default=128, metadata={"help": "the batch size"})
+    mini_batch_size: Optional[int] = field(default=16, metadata={"help": "the PPO minibatch size"})
+    batch_size: Optional[int] = field(default=256, metadata={"help": "the batch size"})
```
Comment (lvwerra) [inline examples/sentiment/scripts/gpt2-sentiment.py:64]:
```suggestion
    mini_batch_size: Optional[int] = field(default=128, metadata={"help": "the PPO minibatch size"})
    batch_size: Optional[int] = field(default=128, metadata={"help": "the batch size"})
```

**Comment by lvwerra** [comment]
Results across 8 seeds - no neg KL issues:

Screenshot 2023-07-03 at 18 54 47

**Review by edbeeching — APPROVED** [review]
Reviewing this as we would like to downstream this change to H4. 

LGTM.
